File: app/src/main/java/piuk/blockchain/android/ui/account/AccountItem.java
package piuk.blockchain.android.ui.account;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

class AccountItem {

    static final int TYPE_CREATE_NEW_WALLET_BUTTON = 0;
    static final int TYPE_IMPORT_ADDRESS_BUTTON = 1;
    static final int TYPE_ACCOUNT = 2;

    private String label;
    private String address;
    private String amount;
    private boolean isArchived;
    private boolean isWatchOnly;
    private boolean isDefault;
    private Integer correctedPosition;
    private int type;

    AccountItem(@Nullable Integer correctedPosition,
                String label,
                String address,
                String amount,
                boolean isArchived,
                boolean isWatchOnly,
                boolean isDefault,
                int type) {
        this.correctedPosition = correctedPosition;
        this.label = label;
        this.address = address;
        this.amount = amount;
        this.isArchived = isArchived;
        this.isWatchOnly = isWatchOnly;
        this.isDefault = isDefault;
        this.type = type;
    }

    AccountItem(int type) {
        this.type = type;
    }

    @NonNull
    public String getLabel() {
        return label != null ? label : "";
    }

    @NonNull
    public String getAddress() {
        return address != null ? address : "";
    }

    @NonNull
    public String getAmount() {
        return amount != null ? amount : "";
    }

    public boolean isArchived() {
        return isArchived;
    }

    public void setIsArchived(boolean isArchived) {
        this.isArchived = isArchived;
    }

    public boolean isWatchOnly() {
        return isWatchOnly;
    }

    public void setIsWatchOnly(boolean isWatchOnly) {
        this.isWatchOnly = isWatchOnly;
    }

    public boolean isDefault() {
        return isDefault;
    }

    public void setDefault(boolean aDefault) {
        isDefault = aDefault;
    }

    public int getType() {
        return type;
    }

    @NonNull
    Integer getCorrectPosition() {
        return correctedPosition != null ? correctedPosition : -1;
    }
}

File: app/src/main/java/piuk/blockchain/android/ui/account/AccountItem.kt
package piuk.blockchain.android.ui.account

class AccountItem {

    var label: String? = null
    var address: String? = null
    var amount: String = ""
    var isArchived: Boolean = false
    var isWatchOnly: Boolean = false
    var isDefault: Boolean = false
    var correctedPosition: Int = -1
    var type: Int = 0

    internal constructor(correctedPosition: Int,
                         label: String?,
                         address: String?,
                         amount: String,
                         isArchived: Boolean,
                         isWatchOnly: Boolean,
                         isDefault: Boolean,
                         type: Int) {
        this.correctedPosition = correctedPosition
        this.label = label
        this.address = address
        this.amount = amount
        this.isArchived = isArchived
        this.isWatchOnly = isWatchOnly
        this.isDefault = isDefault
        this.type = type
    }

    internal constructor(type: Int) {
        this.type = type
    }

    internal constructor(type: Int, amount: String) {
        this.type = type
        this.amount = amount
    }

    companion object {
        const val TYPE_CREATE_NEW_WALLET_BUTTON = 0
        const val TYPE_IMPORT_ADDRESS_BUTTON = 1
        const val TYPE_ACCOUNT_BTC = 2
        const val TYPE_ACCOUNT_BCH = 3
        // Non-clickable types for BCH
        const val TYPE_WALLET_HEADER = 4
        const val TYPE_LEGACY_HEADER = 5
        const val TYPE_LEGACY_SUMMARY = 6
    }
}